premake5.lua

```diff
@@ -217,10 +231,12 @@
 		links       { "lua-lib" }
 
 		-- optional 3rd party libraries
-		if not _OPTIONS["no-zlib"] then
+		filter { "options:zlib-src=contrib" }
 			includedirs { "contrib/zlib", "contrib/libzip" }
 			links { "zip-lib", "zlib-lib" }
-		end
+		filter { "options:zlib-src=system" }
+			links { "zip", "z" }
+		filter {}
```

Comment:
```suggestion
```
There is another filter just after.

Reply:
(Maybe?) Hot take: I like having filters closed after, before opening up another.

Reply:
Consistency with other
either
```
filter A
-- ..
filter {}
filter B
-- ..
filter {}
filter C
-- ..
filter {}
filter D
-- ..
filter{}
```
or
```
filter A
-- ..
filter B
-- ..
filter C
-- ..
filter D
-- ..
filter{}
```

Reply:
Yeah, let's get it consistent first :)

Reply:
FWIW, I went for consistency with the block below which required the `filter{}`. My goal for both blocks was to reduce the changes as much as possible. I was intending to follow up all these changes with another PR to cleanup this up and the premake5.lua scripts in general (if there's anything else to clean up).

I can fix it now if you guys would prefer?

Reply:
If we go for a cleanup PR after, I'm happy either way.